distuv/beta.go

```diff
@@ -0,0 +1,99 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Beta implements the Beta distribution, a two-parameter continuous distribution
+// with support over the positive real numbers.
+//
+// The beta distribution has density function
+//  x^(α-1) * (1-x)^(β-1) * Γ(α+β) / (Γ(α)*Γ(β))
+//
+// For more information, see https://en.wikipedia.org/wiki/Beta_distribution
+type Beta struct {
+	// Alpha is the left shape parameter of the distribution. Alpha must be greater
+	// than 0.
+	Alpha float64
+	// Beta is the right shape parameter of the distribution. Beta must be greater
+	// than 0.
+	Beta float64
+
+	Source *rand.Rand
+}
+
+// ExKurtosis returns the excess kurtosis of the distribution.
+func (be Beta) ExKurtosis() float64 {
+	a := be.Alpha
+	b := be.Beta
+	num := 6 * (math.Pow(a-b, 2)*(a+b+1) - a*b*(a+b+2))
+	den := a * b * (a + b + 2) * (a + b + 3)
+	return num / den
+}
+
+// LogProb computes the natural logarithm of the value of the probability
+// density function at x.
+func (be Beta) LogProb(x float64) float64 {
+	if x < 0 || x > 1 {
+		return math.NaN()
+	}
+	a := be.Alpha
+	b := be.Beta
+	lgab, sign := math.Lgamma(a + b)
```

Comment:
This usage of sign isn't correct. I would add a check that Alpha and Beta are > 0 to the beginning of this function (and panic otherwise). Then, here you can use

````
lab, _ := math.Lgamma(a+b)
la, _ := math.Lgamma(a)
lb, _ := math.Lgamma(b)
return lab - la - lb + (a-1)*math.Log(x) + (b-1)*math.Log(1-x)
````